### PR TITLE
Add instigator to zombie events damaging & dying

### LINKED ISSUE #253: Event:Zombies killed by players
Why is there no event of killing zombies,Will it increase in the future?

**Thanks for your answer.**

### PR DESCRIPTION
**Pull request by DiFFoZ**
Should remove `DropLoot` & `TrackKill` from the event because it will be always true? Fixes #253

### CODE REVIEW

**Comment by IAmSilK** [comment]
> 
> 
> Should remove `DropLoot` & `TrackKill` from the event because it will be always true? Fixes #253

These were left so plugins could modify them.

File: unturned/OpenMod.Unturned/Zombies/Events/ZombieEventsListener.cs [hunk 1]
```diff
@@ -66,34 +67,35 @@ private void Events_OnZombieAlertingPosition(Zombie nativeZombie, ref Vector3 po
...
-            trackKill = @event.TrackKill;
-            dropLoot = @event.DropLoot;
-            stunOverride = @event.StunOverride;
-            cancel = @event.IsCancelled;
+            parameters.damage = @event.DamageAmount;
+            parameters.direction = @event.Ragdoll;
+            parameters.ragdollEffect = @event.RagdollEffect;
+            parameters.instigator = @event.Instigator.Player;
+            // parameters.trackKill = @event.TrackKill;
+            // parameters.dropLoot = @event.DropLoot;
```
Comment (Trojaner) [inline unturned/OpenMod.Unturned/Zombies/Events/ZombieEventsListener.cs:96]: Please uncomment as plugins might want to change this

**Review by Trojaner — CHANGES_REQUESTED** [review]

File: unturned/OpenMod.Unturned/Zombies/Events/ZombieEventsListener.cs (hunk 1 above)
Comment (DiFFoZ) [reply unturned/OpenMod.Unturned/Zombies/Events/ZombieEventsListener.cs:96]: In parameters doesn't have 'trackKill' and 'dropLoot'.

Comment (Trojaner) [reply unturned/OpenMod.Unturned/Zombies/Events/ZombieEventsListener.cs:96]: @IAmSilK

Comment (IAmSilK) [reply unturned/OpenMod.Unturned/Zombies/Events/ZombieEventsListener.cs:96]: These parameters no longer exist as the hook was moved from the direct function to a supported event. I didn't realize this in my original comment.

**Comment by Trojaner** [comment]
Please remove DropLoot & TrackKill, will merge then

**Pull request merged**